py/server/deephaven/experimental/iceberg.py

```diff
@@ -271,9 +272,95 @@ def from_current_snapshot(cls) -> 'SchemaProvider':
         return cls(_JSchemaProvider.fromCurrentSnapshot())
 
 
+class SortOrderProvider(JObjectWrapper):
+    """
+    `SortOrderProvider` is used to specify the sort order for new data when writing to an Iceberg table. More details
+    about sort order can be found in the Iceberg spec: https://iceberg.apache.org/spec/#sorting.
+    Users can specify the sort order in multiple ways, such as by providing a sort ID or using the table's default sort
+    order. This class consists of factory methods to create different sort order providers.
+    """
+
+    j_object_type = _JSortOrderProvider
+
+    def __init__(self, _j_object: jpy.JType):
+        """
+        Initializes the `SortOrderProvider` object.
+
+        Args:
+            _j_object (SortOrderProvider): the Java `SortOrderProvider` object.
+        """
+        self._j_object = _j_object
+
+    @property
+    def j_object(self) -> jpy.JType:
+        return self._j_object
+
+    @classmethod
+    def unsorted(cls) -> 'SortOrderProvider':
+        """
+        Used to disable sorting while writing new data to the Iceberg table.
+
+        Returns:
+            the `SortOrderProvider` object.
+        """
+        return cls(_JSortOrderProvider.unsorted())
+
+    @classmethod
+    def use_table_default(cls) -> 'SortOrderProvider':
+        """
+        Use the default sort order of the table while writing new data. If no sort order is set on the table, no sorting
+        will be done.
+
+        Returns:
+            the `SortOrderProvider` object.
+        """
+        return cls(_JSortOrderProvider.useTableDefault())
+
+    @classmethod
+    def from_sort_id(cls, sort_order_id: int) -> 'SortOrderProvider':
```

Comment:
Jianfeng should comment on the use of `@classmethod` here vs using `@staticmethod`.  The difference may not matter in practice.

Reply:
@classmethod is the go-to way to implement factory methods. So the use of it here it appropriate.